server/sonar-web/src/main/js/apps/quality-gates/components/ThresholdInput.tsx

```diff
@@ -17,23 +17,23 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import Select from '../../../components/controls/Select';
+import { Metric } from '../../../app/types';
 
-export default class ThresholdInput extends React.PureComponent {
-  static propTypes = {
-    name: PropTypes.string.isRequired,
-    value: PropTypes.any,
-    metric: PropTypes.object.isRequired,
-    onChange: PropTypes.func.isRequired
-  };
+interface Props {
+  name: string;
+  value: string;
+  metric: Metric;
+  onChange: (value: string) => void;
+}
 
-  handleChange = e => {
-    this.props.onChange(e.target.value);
+export default class ThresholdInput extends React.PureComponent<Props> {
+  handleChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
+    this.props.onChange(e.currentTarget.value);
   };
 
-  handleSelectChange = option => {
+  handleSelectChange = (option: any) => {
```

Comment:
`option: { value: string }`

Reply:
Like I told you, if I do that I end up with a typing error in the Select, and currently I don't really know what's wrong and how to fix it.